cdap-app-fabric/src/main/java/io/cdap/cdap/internal/operation/OperationRunStore.java

```diff
@@ -83,19 +87,19 @@ public void createOperationRun(OperationRunId runId, OperationRunDetail detail)
   }
 
   /**
-   * Update the metadata of an operation run.
+   * Update the resources of an operation run.
    *
    * @param runId {@link OperationRunId} for the run
-   * @param metadata new metdata of the run
+   * @param resources updated resources for the run
    * @param sourceId the message id which is responsible for the update
    * @throws OperationRunNotFoundException run with id does not exist in namespace
    */
-  public void updateOperationMeta(OperationRunId runId, OperationMeta metadata,
+  public void updateOperationResources(OperationRunId runId, Set<OperationResource> resources,
```

Comment:
Modified as there is not current public api needed tot update the full meta rather individual fields